🚨 [BUG] Anonymous Page 첫 빌드 : 에러 발생

```shell
cd build && make all
make[1]: Entering directory '/workspaces/pintos_22.04_lab_docker/pintos/vm/build'
gcc -E -nostdinc -I../.. -I../../include/lib -I../../include -I../../include/lib/kernel -I../../lib/kernel -P ../../threads/kernel.lds.S > threads/kernel.lds.s
gcc -c ../../threads/init.c -o threads/init.o -g -msoft-float -O0 -fno-omit-frame-pointer -mno-red-zone -mcmodel=large -fno-plt -fno-pic -mno-sse -fno-stack-protector -fno-stack-clash-protection -nostdinc -I../.. -I../../include/lib -I../../include -I../../include/lib/kernel -I../../lib/kernel -Wall -W -Wstrict-prototypes -Wmissing-prototypes -Wsystem-headers -DUSERPROG -DFILESYS -DVM -MMD -MF threads/init.d
In file included from ../../include/threads/thread.h:12,
                 from ../../threads/init.c:23:
../../include/vm/vm.h:126:10: error: ‘hash_hash_func’ redeclared as different kind of symbol
  126 | uint64_t hash_hash_func(const struct hash_elem *e, void *aux);
      |          ^~~~~~~~~~~~~~
In file included from ../../include/vm/vm.h:3,
                 from ../../include/threads/thread.h:12,
                 from ../../threads/init.c:23:
../../include/lib/kernel/hash.h:44:18: note: previous declaration of ‘hash_hash_func’ with type ‘hash_hash_func’ {aka ‘long long unsigned int(const struct hash_elem *, void *)’}
   44 | typedef uint64_t hash_hash_func (const struct hash_elem *e, void *aux);
      |                  ^~~~~~~~~~~~~~
In file included from ../../include/threads/thread.h:12,
                 from ../../threads/init.c:23:
../../include/vm/vm.h:128:6: error: ‘hash_less_func’ redeclared as different kind of symbol
  128 | bool hash_less_func(const struct hash_elem *a, const struct hash_elem *b,
      |      ^~~~~~~~~~~~~~
In file included from ../../include/vm/vm.h:3,
                 from ../../include/threads/thread.h:12,
                 from ../../threads/init.c:23:
../../include/lib/kernel/hash.h:49:14: note: previous declaration of ‘hash_less_func’ with type ‘hash_less_func’ {aka ‘_Bool(const struct hash_elem *, const struct hash_elem *, void *)’}
   49 | typedef bool hash_less_func (const struct hash_elem *a,
      |              ^~~~~~~~~~~~~~
In file included from ../../threads/init.c:36:
../../include/devices/disk.h:26:1: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
   26 | void    register_disk_inspect_intr ();
      | ^~~~
../../threads/init.c: In function ‘main’:
<built-in>: warning: function declared ‘noreturn’ has a ‘return’ statement
make[1]: *** [../../Make.config:38: threads/init.o] Error 1
make[1]: Leaving directory '/workspaces/pintos_22.04_lab_docker/pintos/vm/build'
make: *** [../Makefile.kernel:10: all] Error 2
```

윗 에러를 수정합니다.

--- a/pintos/include/vm/vm.h
+++ b/pintos/include/vm/vm.h
@@ -123,8 +123,8 @@ bool vm_claim_page(void *va);
 enum vm_type page_get_type(struct page *page);
 
 /*project_3*/
-uint64_t hash_hash_func(const struct hash_elem *e, void *aux);
+uint64_t spt_hash_func(const struct hash_elem *e, void *aux);
 
-bool hash_less_func(const struct hash_elem *a, const struct hash_elem *b,
-                    void *aux);
+bool spt_less_func(const struct hash_elem *a, const struct hash_elem *b,
+                   void *aux);
 #endif /* VM_VM_H */

--- a/pintos/vm/vm.c
+++ b/pintos/vm/vm.c
@@ -228,7 +228,7 @@ static bool vm_do_claim_page(struct page *page) {
 /* Initialize new supplemental page table */
 void supplemental_page_table_init(struct supplemental_page_table *spt UNUSED) {
   /** Project 3-Memory Management */
-  hash_init(&spt->spt_hash, hash_hash_func, hash_less_func, NULL);
+  hash_init(&spt->spt_hash, spt_hash_func, spt_less_func, NULL);
 }
 
 bool supplemental_page_table_copy(struct supplemental_page_table *dst,
@@ -242,13 +242,13 @@ void supplemental_page_table_kill(struct supplemental_page_table *spt) {
    * writeback all the modified contents to the storage. */
 }
 
-uint64_t hash_hash_func(const struct hash_elem *e, void *aux) {
+uint64_t spt_hash_func(const struct hash_elem *e, void *aux) {
   struct page *page = hash_entry(e, struct page, hash_elem);
   return hash_bytes(page->va, sizeof *page->va);
 }
 
-bool hash_less_func(const struct hash_elem *a, const struct hash_elem *b,
-                    void *aux) {
+bool spt_less_func(const struct hash_elem *a, const struct hash_elem *b,
+                   void *aux) {
   struct page *page_a = hash_entry(a, struct page, hash_elem);
   struct page *page_b = hash_entry(b, struct page, hash_elem);
 

